app/src/main/java/com/projectthrive/chess/ui/main/PieceMovementRules.kt

```diff
@@ -0,0 +1,69 @@
+package com.projectthrive.chess.ui.main
+
+class PieceMovementRules() {
+    val initialPawnsPositions = mutableSetOf<Position>()
```

Comment:
You can store which pawns have already moved - that would be simpler than dealing with positions